metloom/pointdata/mesowest.py

```diff
@@ -0,0 +1,273 @@
+                    'Z', '')), axis=1)
+        sensor_df["datetime"] = sensor_df["datetime"].apply(self._handle_df_tz)
+        # TODO: Review whether this is necessary?
+        sensor_df['measurementDate'] = sensor_df['datetime'].copy()
+
+        # set index so joinng works
+        sensor_df.set_index("datetime", inplace=True)
+        sensor_df = sensor_df.filter(final_columns)
+        sensor_df[f"{sensor.name}_units"] = response_data['UNITS'][f"{sensor.code}"]
+
+        return sensor_df
+
+    def get_hourly_data(
+        self,
+        start_date: datetime,
+        end_date: datetime,
+        variables: List[SensorDescription],
+    ):
+        """
+        Get hourly measurement data
+        Args:
+            start_date: datetime object for start of data collection period
+            end_date: datetime object for end of data collection period
+            variables: List of metloom.variables.SensorDescription object
+                from self.ALLOWED_VARIABLES
+        Returns:
+            GeoDataFrame of data. The dataframe should be indexed on
+            ['datetime', 'site'] and have columns
+            ['geometry', 'site', 'measurementDate']. Additionally, for each
+            variables, it should have column f'{variable.name}' and
+            f'{variable.name}_UNITS'
+            See CDECPointData._get_data for example implementation and
+            TestCDECStation.tny_daily_expected for example dataframe.
+            Datetimes should be in UTC
+        """
+        df = self._get_data(start_date, end_date, variables, interval='H')
+        return df
+
+    def get_daily_data(
+        self,
+        start_date: datetime,
+        end_date: datetime,
+        variables: List[SensorDescription],
+    ):
+        """
+        Get daily measurement data
+        Args:
+            start_date: datetime object for start of data collection period
+            end_date: datetime object for end of data collection period
+            variables: List of metloom.variables.SensorDescription object
+                from self.ALLOWED_VARIABLES
+        Returns:
+            GeoDataFrame of data. The dataframe should be indexed on
+            ['datetime', 'site'] and have columns
+            ['geometry', 'site', 'measurementDate']. Additionally, for each
+            variables, it should have column f'{variable.name}' and
+            f'{variable.name}_UNITS'
+            See CDECPointData._get_data for example implementation and
+            TestCDECStation.tny_daily_expected for example dataframe.
+            Datetimes should be in UTC
+        """
+        df = self._get_data(start_date, end_date, variables, interval='D')
+        return df
+
+    @classmethod
+    def points_from_geometry(
+        cls,
+        geometry: gpd.GeoDataFrame,
+        variables: List[SensorDescription],
+        snow_courses=False,
+        within_geometry=True,
+        token_json="~/.synoptic_token.json"
```

Comment:
You could add a `**kwargs` to this and all the other methods as a way of avoiding inconsistencies in the implementation?